File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        s = [(temperatures[-1],n-1)]
        results = [0]*n
        for i in range(n-2,-1,-1):
            t = temperatures[i]
            top = s[-1][0]
            while(len(s) and s[-1][0]<=t):
                s.pop()
            if not len(s):
                results[i]=0
            else:
                results[i]=s[-1][1]-i
            s.append((t,i))
        return results

            

File: Data Structures & Algorithms/minimum-interval-including-query/submission-1.cpp
class Solution {
public:
    static bool comp(vector<int> &a,vector<int> &b){
        if(a[0]==b[0]) return a[1]<b[1];
        return a[0]<b[0]; 
    }
    vector<int> minInterval(vector<vector<int>>& intervals, vector<int>& queries) {
        sort(intervals.begin(),intervals.end(),comp);

        vector<pair<int,int>> sortedq;
        for(int i=0;i<queries.size();i++) sortedq.push_back({queries[i],i});
        sort(sortedq.begin(),sortedq.end());

        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>> heap;
        vector<int> ans(queries.size());
        int i=0;

        for(auto &[q,ind] :sortedq){
            while(i<intervals.size() && intervals[i][0]<=q){
                heap.push({intervals[i][1]-intervals[i][0]+1,intervals[i][1]});
                i++;
            }
            while(!heap.empty() && heap.top().second<q) heap.pop();
            ans[ind]=heap.empty()?-1:heap.top().first;
        }
        return ans;
    }
};
